lizmap/modules/view/controllers/media.classic.php

```diff
@@ -287,11 +287,34 @@ public function getMedia()
             $rep->fileName = '';
             $rep->content = $content;
         }
+        // For HEAD request, we need to set the content length
+        $fileSize = filesize($finalPath);
+        $rep->addHttpHeader('Content-Length', $fileSize);
 
-        $rep->setExpires('+1 days');
+        // Set Allow header to GET and HEAD for all files
+        $rep->addHttpHeader('Allow', 'GET, HEAD');
```

Comment:
This header is only useful when you return an error 405 Method Not Allowed, after checking the HTTP method.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Allow

```
if ($_SERVER['REQUEST_METHOD'] != 'GET' && $_SERVER['REQUEST_METHOD'] != 'HEAD') {
    $rep = $this->getResponse(....);
    $rep->setHttpStatus(405, 'Method not Allowed');
    $rep->addHttpHeader('Allow', 'GET, HEAD');
    return $rep;
}
```